modules/data.remote/R/SMAP_SMP_prep.R

```diff
@@ -0,0 +1,95 @@
+#' Prepare SMAP Soil Moisture (SMP) data for the SDA workflow.
+#'
+#' @param site_info Bety list of site info including site_id, lon, and lat.
+#' @param start_date Start date of SDA workflow.
+#' @param end_date End date of SDA workflow.
+#' @param time_points A vector contains each time point within the start and end date.
+#' @param outdir Where the final CSV file, and the CSV file from GEE are stored.
+#' @param search_window search window for locate available SMP values.
+#' @param export_csv Decide if we want to export the CSV file.
+#' @param update_csv Decide if we want to update current CSV file given an updated SMAP_gee.csv file
+#'
+#' @return A data frame containing SMAP smp and sd for each site and each time step.
+#' @export
+#'
+#' @examples
+#' @author Dongchen Zhang
+#' @importFrom magrittr %>%
+SMAP_SMP_prep <- function(site_info, start_date, end_date, time_points, 
+                      outdir, search_window = 30, export_csv = TRUE, update_csv = FALSE){
+  #note that, the SMAP_gee.csv file comes from Google Earth Engine (GEE) directly.
+  #Code for generating this file can be found through this link: 
+  #https://code.earthengine.google.com/ecbeb770e576d8ef72f72f5f12da3496
+  #Feel free to contact Dongchen Zhang (zhangdc@bu.edu) who wrote this code.
+  #The SMAP.csv file will be generated the first time when you use this code.
+  #for the next time, it will save you lot of time if you can provide the SMAP.csv directly.
+  
+  #check if SMAP.csv exists.
+  if(!file.exists(file.path(outdir, "SMAP.csv")) | as.logical(update_csv)){
+    if(!file.exists(file.path(outdir, "SMAP_gee.csv"))){
+      PEcAn.logger::logger.info("Please Provide SMAP dir that contains at least the SMAP_gee.csv file!")
+      return(0)
+    }else{
+      SMAP_gee <- utils::read.csv(file.path(outdir, "SMAP_gee.csv"))
+      
+      #prepare CSV from Current SMAP_gee file
+      SMAP_CSV <- matrix(NA, 0, 6) %>% `colnames<-`(c("date", "site_id", "lat", "lon", "smp", "sd"))
+      
+      for (i in 2 : dim(SMAP_gee)[1]) {
+        String <-  strsplit(gsub(",", "", gsub("\\[|\\]", "", SMAP_gee[i,2])), " ")[[1]]
+        date <- as.Date(strsplit(String[1], "_")[[1]][5], "%Y%m%d")
+        lon <- as.numeric(String[2])
+        lat <- as.numeric(String[3])
+        smp <- as.numeric(String[5]) * 100
+        sd <- 0.04 * 100 #From Daniel
+        
+        #Match current lon/lat with site_info
+        Distance <- swfscMisc::distance(lat1 = lat, lon1 = lon, lat2 = site_info$lat, lon2 = site_info$lon)
```

Comment:
I haven't looked through the full list of PEcAn dependencies, but I think the following functions all have a great circle distance function:
* codep
* geosphere
* PBSmapping
* lmap
* gdistance

Reply:
didn't see any of these in pecan.depends.R but I did find geosphere in my workspace on the lab server, which makes me suspicious that it might be a dependency of something in pecan (though nothing jumped out at me on their reverse depends/imports list)

here are some other suggestions that are along the lines of what @infotroph was suggesting:
terra::distance
sp::spDists or spDistsN1

Reply:
I would keep this for now, and if we come up with some existing dependencies, we could update the code and use it.